express/component_spec.py

```diff
@@ -0,0 +1,335 @@
+"""This module defines classes to represent an Express component specification."""
+import copy
+import json
+import yaml
+import pkgutil
+import types
+import typing as t
+from dataclasses import dataclass, asdict
+
+import jsonschema.exceptions
+from jsonschema import Draft4Validator
+
+from express.exceptions import InvalidComponentSpec
+from express.io_utils import load_yaml
+from express.common import Field
+from express.pipeline_utils import get_kubeflow_type
+
+
+@dataclass
+class Input:
+    """
+    Component input arguments
+    Attributes:
+        name: name of the argument
+        description: argument description
+        type: the python argument type (str, int, ...)
+    """
+
+    name: str
+    description: str
+    type: str
+
+
+@dataclass
+class Output:
+    """
+    Component output argument
+    Attributes:
+        name: name of the argument
+        description: argument description
+    """
+
+    name: str
+    description: str
+
+
+T = t.TypeVar("T", Input, Output)
+
+
+@dataclass
+class KFPComponent:
+    """
+    A class representing a Kubeflow Pipelines component.
+    Attributes:
+        name: The name of the component.
+        description: The description of the component.
+        image: The Docker image for the component.
+        args: A dictionary of component arguments.
+        inputs: The input parameters for the component.
+        outputs: The output parameters for the component.
+        command: The command to run the component.
+    """
+
+    name: str
+    description: str
+    image: str
+    args: t.Dict[str, t.Dict[str, str]]
+    inputs: t.Union[Input, t.List[Input]] = None
+    outputs: t.Union[Output, t.List[Output]] = None
+    command: t.List[t.Union[str, t.Dict[str, str]]] = None
+
+    def __post_init__(self):
+        self.inputs = self._add_defaults_values(
+            self.inputs, self._get_default_inputs(), Input
+        )
+
+        self.outputs = self._add_defaults_values(
+            self.outputs, self._get_default_outputs(), Output
+        )
+
+        self._add_input_args()
+        self._set_component_run_cmd()
+
+    @staticmethod
+    def _get_default_inputs() -> t.List[Input]:
+        """
+        Set default inputs to components
+        """
+        inputs = [
+            Input(
+                name="input_manifest",
+                description="Path to the the input manifest",
+                type="String",
+            ),
+            Input(
+                name="extra_args",
```

Comment:
```suggestion
                name="args",
```
Let's just call it args. Does this include the metadata like run_id and component_id?